contracts/adapters/SwapAdapter.sol

```diff
@@ -411,23 +415,28 @@ contract SwapAdapter is AccessControl {
         address tokenIn,
         address tokenOut,
         uint256 amountInMaximum,
-        uint256 amountOut,
-        uint256 valueToSend
+        uint256 amountOut
     ) internal returns(uint256 amountIn) {
+        uint256 balanceBefore = IERC20(tokenIn).balanceOf(address(this));
         bytes memory path = _verifyAndEncodePath(
             pathTokens,
             pathFees,
             tokenIn,
             tokenOut
         );
-        IV3SwapRouter.ExactOutputParams memory params = IV3SwapRouter.ExactOutputParams({
-            path: path,
-            recipient: address(this),
-            amountOut: amountOut,
-            amountInMaximum: amountInMaximum
-        });
-
-        amountIn = _swapRouter.exactOutput{value: valueToSend}(params);
+        IPermit2(_permit2).approve(tokenIn, address(_swapRouter), uint160(amountInMaximum), uint48(block.timestamp));
+        bytes memory commands = abi.encodePacked(uint8(0x01)); // V3_SWAP_EXACT_OUT
```

Comment:
Adding the `uint8 V3_SWAP_EXACT_OUT = 1;` constant could improve readability and make the comment unnecessary.